[JohnYoo] iP

johnyoozhengxian opened the pull request:
# TalkingJohn
## TalkingJohn frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

## All you need to do is,

1. download it from [here](https://github.com/johnyoozhengxian/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

## And it is **FREE**! (paid version with GUI COMING SOON)

## Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

### If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(TalkingJohn.class, args);
    }
}```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

johnyoozhengxian committed:
Implemented TalkingJohn Level0

johnyoozhengxian committed:
Implemented TalkingJohn Level0

johnyoozhengxian committed:
Implemented TalkingJohn Level1

johnyoozhengxian committed:
Implemented TalkingJohn Level2

johnyoozhengxian committed:
Implemented TalkingJohn Level3

johnyoozhengxian committed:
Implemented TalkingJohn Level4

johnyoozhengxian committed:
Implemented TalkingJohn Level5

johnyoozhengxian committed:
Implemented TalkingJohn Level6

johnyoozhengxian committed:
Implemented testing

johnyoozhengxian committed:
Implemented branch + Level7

johnyoozhengxian committed:
Merge branch-Level-7 into master

johnyoozhengxian committed:
Merge branch-Level-7 into master

johnyoozhengxian committed:
Merge branch 'branch-Level-7'

johnyoozhengxian committed:
Implemented Level-8

johnyoozhengxian committed:
Implemented better OOP

johnyoozhengxian committed:
Implemented Packages

johnyoozhengxian committed:
Merge branch 'add-gradle-support'

johnyoozhengxian committed:
Implemented Gradle

johnyoozhengxian committed:
Implemented JUnit

johnyoozhengxian committed:
Implemented jar

johnyoozhengxian committed:
implemented javadocs

johnyoozhengxian committed:
Implemented Coding Standards

johnyoozhengxian committed:
find function added

chaaaaun commented on src/main/java/TalkingJohn/Event.java:
```diff
@@ -0,0 +1,33 @@
+ * Represents an event task that has a start date and an end date.
+ */
+public class Event extends Task {
+    protected String by1;
```
It might be easy to confuse these 2 variables, maybe you could use `from` and `to` or `start` and `end` or something similar?

chaaaaun commented on src/main/java/TalkingJohn/Parser.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * List of invalid inputs that do not trigger any action.
+     */
+    private final List<String> invalidInputs = Arrays.asList("todo", "deadline",
```
This seems like a list of valid inputs to me?

chaaaaun commented on src/main/java/TalkingJohn/TaskList.java:
```diff
@@ -0,0 +1,121 @@
+        } else if (input.startsWith("mark") && input.length() > 4) {
+            try {
+                int i = convertInt(input);
+                Task taskToMark = taskArr.get(i - 1);
```
Like the naming here, very clear.

chaaaaun commented on src/test/java/TalkingJohn/TJohnTest.java:
```diff
@@ -0,0 +1,21 @@
+public class TJohnTest {
+
+    @Test
+    public void testEventToString() {
```
Can consider using the snake case format to name your tests, so that it is clear from the start what the input and expected output is.

chaaaaun commented on src/main/java/TalkingJohn/TaskList.java:
```diff
@@ -0,0 +1,121 @@
+                String[] parts = input.split(" ", 2);
+                String buffer = parts[1];
+                String[] secPart = buffer.split("/", 3);
+                Event event = new Event(secPart[0], secPart[1], secPart[2]);
```
Could use a more descriptive name, didn't expect your `secPart` to split into more parts.

chaaaaun reviewed (commented):
Nice work! Most of your code is idomatic and easy to read.

johnyoozhengxian committed:
Implemented TalkingJohn Level10

johnyoozhengxian committed:
Add Assertions

There are several assertions that was added in
for debugging and testing purposes, serving
as sanity checks to ensure that certain conditions
hold true during the execution of a program.

johnyoozhengxian committed:
Add CodeQuality

There are several code quality issues for this
project, which I solved by removing most of the
nested loops, as well as changing some variable
names, and limiting the function length to 30
lines.

johnyoozhengxian committed:
Merge pull request #3 from johnyoozhengxian/branch-A-CodeQuality

Branch a code quality

johnyoozhengxian committed:
Add Notes

Add Notes. Users now can add notes to
Tasks they add by using ":" after the description
and before the Deadline and Event timings.

johnyoozhengxian committed:
Add User-Guide

johnyoozhengxian committed:
Add User-Guide

johnyoozhengxian committed:
Add JavaDocs
